compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -775,6 +810,15 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
     res
   }
 
+  def checkGADT(tp: Type): Boolean = {
```

Comment:
I'm not sure what you mean by `checkGADT` the implementation looks like it's computing `isDeeplyInvariant`

Reply:
Yes, something like `shouldCheckGADT` with a comment is more friendly.